test/debug/session_test.rb

```diff
@@ -17,15 +17,15 @@ def program
     end
 
     def test_session_starts_manually
-      debug_code(program, boot_options: "") do
+      debug_code(program, boot_options: "", remote: false) do
```

Comment:
@ono-max why is `remote: false` needed here? changing `boot_options` shouldn't start remote debugging anyway?

Reply:
Because this test doesn't need to be executed as remote.

> changing boot_options shouldn't start remote debugging anyway?

I don't get it. If I run this test without `remote: false`, it failed. Could you tell me more?

Reply:
@st0012 
Because this test doesn't need to be executed as remote.

> changing boot_options shouldn't start remote debugging anyway?

I don't get it. If I ran this test without `remote: false`, it failed. Could you tell me more?

```shell
ruby test/debug/session_test.rb
Loaded suite test/debug/session_test
Started
.F
================================================================================
Failure: test_session_starts_manually(DEBUGGER__::ConsoleStartTest):
  Expected line number to be 5, but was 1

  [DEBUG SESSION LOG]
  > [1, 7] in /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210701-11874-3zc8c.rb
  > =>    1| a = 1
  >       2| b = 2
  >       3| require "debug"
  >       4| DEBUGGER__.console
  >       5| c = 3
  >       6| binding.bp
  >       7| "foo"
  > =>#0	<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210701-11874-3zc8c.rb:1
/Users/ono-max/workspace/debug/test/support/assertions.rb:8:in `block in assert_line_num'
/Users/ono-max/workspace/debug/test/support/utils.rb:87:in `block (2 levels) in create_pseudo_terminal'
/Users/ono-max/.rbenv/versions/3.0.0/lib/ruby/3.0.0/timeout.rb:97:in `block in timeout'
/Users/ono-max/.rbenv/versions/3.0.0/lib/ruby/3.0.0/timeout.rb:35:in `block in catch'
/Users/ono-max/.rbenv/versions/3.0.0/lib/ruby/3.0.0/timeout.rb:35:in `catch'
/Users/ono-max/.rbenv/versions/3.0.0/lib/ruby/3.0.0/timeout.rb:35:in `catch'
/Users/ono-max/.rbenv/versions/3.0.0/lib/ruby/3.0.0/timeout.rb:112:in `timeout'
/Users/ono-max/workspace/debug/test/support/utils.rb:79:in `block in create_pseudo_terminal'
/Users/ono-max/workspace/debug/test/support/utils.rb:75:in `spawn'
/Users/ono-max/workspace/debug/test/support/utils.rb:75:in `create_pseudo_terminal'
/Users/ono-max/workspace/debug/test/support/utils.rb:58:in `setup_terminal'
/Users/ono-max/workspace/debug/test/support/utils.rb:45:in `setup_terminal'
/Users/ono-max/workspace/debug/test/support/utils.rb:26:in `debug_code'
test/debug/session_test.rb:20:in `test_session_starts_manually'
     17:     end
     18:
     19:     def test_session_starts_manually
  => 20:       debug_code(program, boot_options: "") do
     21:         assert_line_num(5)
     22:         type 'quit'
     23:         type 'y'
================================================================================
..
Finished in 0.83998 seconds.
--------------------------------------------------------------------------------
4 tests, 9 assertions, 1 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
75% passed
--------------------------------------------------------------------------------
4.76 tests/s, 10.71 assertions/s
```

Reply:
sorry that I initially thought it's a toggle like choosing either local or remote mode. but it looks like all tests are now tested in both local and remote mode, and `remote: false` is to skip the remote part, right?

if that's the case, can you document this behavior (`debug_code` runs both local and remote mode) above the `debug_code` method?

Reply:
> all tests are now tested in both local and remote mode, and remote: false is to skip the remote part, right?

Exactly 😄 

> if that's the case, can you document this behavior (debug_code runs both local and remote mode) above the debug_code method?

Sure.

Reply:
https://github.com/ruby/debug/pull/76/files#diff-ee14bf65f14927f7778a246f90c5cec0c581be3e2658c2a8a3726c59db5be3f9R21